js/shortcode-quicktag-button.js

```diff
@@ -103,7 +103,9 @@ window.wp_sc_buttons.qt = (function(window, document, $, QTags, buttons, scbutto
 
 		btn.buildShortCode = function( shortcode_params ) {
 
-			var shortcode = '['+ params.slug;
+			var shortcode = '['+ params.slug,
+				close_tag = '[/' + params.slug + ']',
+				mce_selection = ( tinymce.activeEditor ) ? tinymce.activeEditor.selection.getContent() : false;
```

Comment:
Does this work on/in the text view? I know there's a way to get the selection in the text view as well.